mat64/vector.go

```diff
@@ -188,48 +188,30 @@ func (v *Vector) DivElemVec(a, b *Vector) {
 	}
 }
 
-// MulVec computes a * b if trans == false and a^T * b if trans == true. The
-// result is stored into the receiver. MulVec panics if the number of columns in
-// a does not equal the number of rows in b.
-func (v *Vector) MulVec(a Matrix, trans bool, b *Vector) {
-	ar, ac := a.Dims()
+// MulVec computes a * b. The result is stored into the receiver.
+// MulVec panics if the number of columns in a does not equal the number of rows in b.
+func (v *Vector) MulVec(aMat Matrix, b *Vector) {
+	r, c := aMat.Dims()
 	br := b.Len()
-	if trans {
-		if ar != br {
-			panic(ErrShape)
-		}
-	} else {
-		if ac != br {
-			panic(ErrShape)
-		}
-	}
-
-	var w Vector
-	if v != a && v != b {
-		w = *v
+	if c != br {
+		panic(ErrShape)
 	}
-	if w.n == 0 {
-		if trans {
-			w.mat.Data = use(w.mat.Data, ac)
-		} else {
-			w.mat.Data = use(w.mat.Data, ar)
-		}
-
-		w.mat.Inc = 1
-		w.n = ar
-		if trans {
-			w.n = ac
-		}
+	a, trans := untranspose(aMat)
+	ar, ac := a.Dims()
+	v.reuseAs(r)
+	var w *Vector
```

Comment:
Will this go away and we use v as you have done in the Dense equivalents? Presumably in the same PR that changes aMat.